public/app/core/components/Login/LoginCtrl.tsx

```diff
@@ -0,0 +1,161 @@
+import React from 'react';
+import appEvents from 'app/core/app_events';
+import config from 'app/core/config';
+
+import { updateLocation } from 'app/core/actions';
+import { connect } from 'react-redux';
+import { StoreState } from 'app/types';
+import { PureComponent } from 'react';
+import { getBackendSrv } from '@grafana/runtime';
+import { hot } from 'react-hot-loader';
+
+const isOauthEnabled = () => Object.keys(config.oauth).length > 0;
+
+export interface FormModel {
+  user: string;
+  password: string;
+  email: string;
+}
+interface Props {
+  routeParams?: any;
+  updateLocation?: typeof updateLocation;
+  children: (obj: {
+    isLoggingIn: boolean;
+    changePassword: (pw: string) => void;
+    isChangingPassword: boolean;
+    skipPasswordChange: Function;
+    login: (data: FormModel) => void;
+    disableLoginForm: boolean;
+    ldapEnabled: boolean;
+    authProxyEnabled: boolean;
+    disableUserSignUp: boolean;
+    isOauthEnabled: boolean;
+    loginHint: string;
+    passwordHint: string;
+  }) => JSX.Element;
+}
+
+interface State {
+  isLoggingIn: boolean;
+  isChangingPassword: boolean;
+}
+
+export class LoginCtrl extends PureComponent<Props, State> {
+  result: any = {};
+  constructor(props: Props) {
+    super(props);
+    this.state = {
+      isLoggingIn: false,
+      isChangingPassword: false,
+    };
+    if (config.loginError) {
+      appEvents.emit('alert-warning', ['Login Failed', config.loginError]);
```

Comment:
uhm, so what's up with this? Is this needed at all?

Reply:
Yeah, I have no idea. I'll remove it, can't find anything about `loginError` in the config.

Reply:
Ok, I see it's set by grafana server and returned via boot data (see pkg/api/login.go). @papagian can u elaborate on that a little bit?

Reply:
`loginError` is returned by the view if OAuth login fails for any reason in order to be displayed to the user the cause of the failure like [this](https://user-images.githubusercontent.com/23303904/60657611-3430dd00-9e84-11e9-8c79-929aeb92d1f6.png).